types/react-table/index.d.ts

```diff
@@ -167,6 +167,10 @@ export interface UseTableColumnOptions<D extends object>
 
 type UpdateHiddenColumns<D extends object> = (oldHidden: Array<IdType<D>>) => Array<IdType<D>>;
 
+export interface TableToggleHideAllColumnProps {
```

Comment:
These default props are identical to `TableToggleRowsSelectedProps` and `TableToggleAllRowsSelectedProps`.
How about making a common interface then? Something like `TableToggleCommonProps`